aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -209,6 +223,15 @@ module aptos_framework::account {
         account_resource.authentication_key = new_auth_key;
     }
 
+    // Check account_pk_bytes matches current_auth_key:
+    //  1. First, append the Ed25519 scheme identifier '0x00' to `account_pk_bytes`
+    //  2. Second, hash this using SHA3-256
+    public fun verify_authentication_key_matches_public_key(account_auth_key: vector<u8>, account_pk_bytes: vector<u8>) : bool {
+        vector::push_back(&mut account_pk_bytes, 0);
+        let expected_account_auth_key = hash::sha3_256(account_pk_bytes);
+        create_address(expected_account_auth_key) == create_address(account_auth_key)
```

Comment:
you don't need to call create_address to make the comparison. just compare them directly